### PR TITLE
CSV reports

### PR DESCRIPTION
**Pull request by jaleszek**
Implementation of structures for all reports.
Implementation of data picking for `registered users` CSV export.

### CODE REVIEW

File: app/models/reports/form_answer.rb [hunk 1]
```diff
@@ -90,6 +90,18 @@ def business_sector_other
...
+  def business_region
+    if business_form?
+      doc "principal_address_region"
+    else
+      doc "nominee_personal_address_region"
+    end
+  end
+
+  def employees
+    doc("employees_5of5").presence || doc("employees_2of2")
```
Comment (RuslanHamidullin) [inline app/models/reports/form_answer.rb]: @jaleszek for International Trade can be 6 or 3, so need to correct I guess
https://github.com/bitzesty/qae/blob/master/app/forms/qae_2014_forms/international_trade/international_trade_step3.rb#L73

**Comment by RuslanHamidullin** [comment]
@jaleszek other stuff looks good, but need to check Trade Form note above ^

File: app/models/reports/form_answer.rb [hunk 2]
```diff
@@ -90,6 +90,22 @@ def business_sector_other
...
+  def business_region
+    if business_form?
+      doc "principal_address_region"
+    else
+      doc "nominee_personal_address_region"
+    end
+  end
+
+  def employees
+    unless trade?
```
Comment (matthewford) [inline app/models/reports/form_answer.rb]: Do not use `unless` with `else`. Rewrite these with the positive case first.

**Comment by jaleszek** [comment]
@RuslanHamidullin nice, thanks, fixed.

**Comment by RuslanHamidullin** [comment]
@jaleszek cool, waiting for travis

File: app/models/reports/form_answer.rb [hunk 3]
```diff
@@ -90,6 +90,22 @@ def business_sector_other
...
+      doc("employees_5of5").presence || doc("employees_2of2")
```
Comment (dreamfall) [inline app/models/reports/form_answer.rb]: Not sure if this is a right check. If user filled stat for 5/6 years first, and then changed it to 2 years `employees_5of5` won't be cleared.

Comment (RuslanHamidullin) [inline app/models/reports/form_answer.rb]: @jaleszek looks like @dreamfall cacthed another issue.
Yeah, if user changed from 2 to 5 so on, will be both in document

File: spec/models/reports/registered_users_spec.rb [hunk 4]
```diff
@@ -1,9 +1,9 @@
 require "rails_helper"
 
-describe Reports::AdminReportBuilder do
+describe Reports::RegisteredUsers do
   let!(:form_answer) { create(:form_answer) }
 
-  let(:output) { described_class.new.build }
+  let(:output) { described_class.new(Settings.current).build }
```
Comment (dreamfall) [inline spec/models/reports/registered_users_spec.rb]: You're passing `Settings` object here, but code uses `AwardYear` object.

File: app/models/reports/form_answer.rb (hunk 3 above)
Comment (RuslanHamidullin) [inline app/models/reports/form_answer.rb]: So need to check selected value for each type of form.
1) Trade 
 "trade_commercial_success" :
```
3 to 5
6 plus
```
2) Sustainable dev form:
"development_performance_years"
```
2 to 4
5 plus
```
3) Innovation
"innovation_performance_years"
```
2 to 4
5 plus
```

**Comment by jaleszek** [comment]
to fix

**Pull request closed without merging**